RIOT-OS/RIOT
# net/sock_dtls: add public key verification

### Contribution description

This adds a pseudomodule to enable public key verification of peers. When `sock_dtls_verify_public_key` is enabled the public key sent by the peer is checked against the registered keys on the corresponding sock. The implementation for tinydtls is added, as well as the corresponding changes to use it on the `dtls-sock` example.

### Testing procedure
Run `examples/dtls-sock` with ECC enabled, the public key should be verified. You can try removing the server's public key from the client, the connection should be aborted.

### Issues/PRs references
Loosely related to #19838
